packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -485,13 +493,45 @@ function runInstructionSync(
     case inst.InstructionKind.Answer:
       runAnswerInstruction(instruction, context);
       break;
+    case inst.InstructionKind.Call:
+      runCallInstruction(instruction, context);
+      break;
     case inst.InstructionKind.Halt:
       runHaltInstruction(instruction, context);
       break;
   }
   return undefined;
 }
 
+function runCallInstruction(
+  instruction: inst.CallInstruction,
+  context: InterpreterContext,
+): void {
+  const procedure = context.procedures.get(instruction.procedureName);
+  if (!procedure) {
+    context.diagnostics.push(
+      diagnosticFromCode(Severe.IBM3968I, instruction.procedureNameToken),
+    );
+    return;
+  }
+  if (procedure.statement) {
+    context.diagnostics.push(
+      diagnosticFromCode(Severe.IBM3971I, instruction.procedureNameToken),
+    );
+  }
+  if (
+    procedure.options.some(
+      (option) => option.kind === ast.SyntaxKind.ReturnsOption,
+    )
+  ) {
+    context.diagnostics.push(
+      diagnosticFromCode(Severe.IBM3970I, instruction.procedureNameToken),
+    );
+  }
```

Comment:
Suggestion: I believe that all of these should be handled by the type system. See reason above.